openfass-go/functions.go

```diff
@@ -0,0 +1,222 @@
+package openfass_go
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"github.com/vitwit/go-fass/rest"
+
+)
+
+const host = "http://127.0.0.1:8080"
+
+// Create a system function
+func (cl *Client) CreateSystemFunctions(data FunctionDefintion) (*rest.Response, error) {
+	//request := GetRequest("/system/functions", "", user, password)
+	cl.Method = "POST"
+	cl.BaseURL = host + "/system/functions"
+	cl.Body = GetRequestBody(data)
```

Comment:
Can you create a utility function something like 
```
CreatePostRequest/GetRequest (cl *Client, method string, url string, data string) {
    cl.Method = method
    cl.BaseUrl = GATEWAY_URL + url
    cl.Body = data
}
```
And you can use something like this everywhere else,

```
    utils.CreateGetRequest(cl, "GET", "/some/url/like/this", "")

    return MakeRequest(cl.Request)
```

? So you can just request that funciton to